lib/src/event_controller.dart

```diff
@@ -344,5 +350,13 @@ class CalendarData<T extends Object?> {
     }
     return events;
   }
+
+  /// Remove all events from the controller.
+  void clear() {
+    _fullDayEventList.clear();
+    _rangingEventList.clear();
+    _singleDayEvents.clear();
```

Comment:
Cleare data from _recurringEventsList as well.

Take latest pull from master to get this list.